### PR TITLE
Implement "attach-mode"

### PR DESCRIPTION
**Pull request by Leon-Plickat**
Solves #90 

More modes (like "below focus" or "above focus") can be easily added in a future commit.

### CODE REVIEW

File: river/Output.zig [hunk 1]
```diff
@@ -107,6 +114,8 @@ pub fn init(self: *Self, root: *Root, wlr_output: *c.wlr_output) !void {
 
     self.layout = try std.mem.dupe(util.gpa, u8, "full");
 
+    self.attach_mode = AttachMode.AttachTop;
```
Comment (ifreund) [inline river/Output.zig:117]: can use an enum literal here
```suggestion
    self.attach_mode = .top;
```

File: river/command/attach_mode.zig [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    allocator: *std.mem.Allocator,
+    seat: *Seat,
+    args: []const []const u8,
+    out: *?[]const u8,
+) Error!void {
+    if (args.len < 2) return Error.NotEnoughArguments;
+    if (args.len > 2) return Error.TooManyArguments;
+
+    if (std.mem.eql(u8, "top", args[1])) {
+        seat.focused_output.attach_mode = Output.AttachMode.AttachTop;
```
Comment (ifreund) [inline river/command/attach_mode.zig:36]: Here as well, can also drop the `Output` import.
```suggestion
        seat.focused_output.attach_mode = .top;
```

File: doc/riverctl.1.scd [hunk 3]
```diff
@@ -95,6 +95,9 @@ that tag 1 through 9 are visible.
 
 ## CONFIGURATION COMMANDS
 
+*attach-mode* *top*|*bottom*
+	Configure where new views should attach into the view stack.
```
Comment (ifreund) [inline doc/riverctl.1.scd:99]:
```suggestion
	Configure where new views should attach to the view stack for the currently focused output.
```

**Comment by Leon-Plickat** [comment]
Maybe it would make more sense to have the mode switching be part of the `viewstack.push()` function instead of different kinds of having multiple push functions and switching every time  a view is attached?

**Comment by ifreund** [comment]
> Maybe it would make more sense to have the mode switching be part of the viewstack.push() function instead of having multiple push functions and switching every time a view is attached?

There are other places that `ViewStack.push()` is used (e.g. the zoom command or the focus stack of seats). Putting this logic in a new `ViewStack.attach()` function would make sense though imo.

File: river/command/attach_mode.zig [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+const std = @import("std");
+
+const util = @import("../util.zig");
+
+const Error = @import("../command.zig").Error;
+const Seat = @import("../Seat.zig");
+const Output = @import("../Output.zig");
```
Comment (ifreund) [inline river/command/attach_mode.zig:24]:
```suggestion
```

**Comment by Leon-Plickat** [comment]
Renamed `viewstack.push_bottom()` to `viewstack.append()` which is a better name IMO.

Also added `viewstack.attach()`, which now handles the switch. The idea is to call `.attach()` when the way a node is attached can change during runtime and to use `.push()` and the other specific attach methods when the mode is hardcoded anyway for whatever reason (as with the zoom command).

**Pull request merged**